src/set.rs

```diff
@@ -249,6 +257,39 @@ mod tests {
     const TEST_BATCH_SMALL: usize = 100;
     const TEST_REP: usize = 20;
 
+    #[test]
+    fn debug_when_empty() {
+        let set: ConSet<String> = ConSet::new();
+        assert_eq!("{}".to_owned(), format!("{:?}", set));
+    }
+
+    #[test]
+    fn debug_when_has_elements() {
+        let set: ConSet<&str> = ConSet::new();
+        assert!(set.insert("hello").is_none());
+        assert!(set.insert("world").is_none());
+        let expected = "{\"hello\", \"world\"}".to_owned();
+        let actual = format!("{:?}", set);
+
+        let mut expected_chars: Vec<char> = expected.chars().collect();
+        expected_chars.sort();
+        let mut actual_chars: Vec<char> = actual.chars().collect();
+        actual_chars.sort();
+        assert_eq!(expected_chars, actual_chars);
```

Comment:
Quite interesting way to test it :-).

This code brings some related ideas to mind, though I don't think either one is strictly better than this, so there's no need to change anything here:

* The `itertools` has nice method [`sorted`](https://docs.rs/itertools/0.8.0/itertools/trait.Itertools.html#method.sorted). It would make the code slightly shorter than the present, but would require bringing the dependency in for the tests ‒ so it's probably not worth it just for this.
* Alternative to sorting the characters would be to simply check something like: `assert!(actual.contains("\"hello\"))`.

Reply:
>     * Alternative to sorting the characters would be to simply check something like: `assert!(actual.contains("\"hello\"))`.

Yep, I was thinking also about something like that or to progressively "remove" from the debug string the elements of the set, ending up with an empty struct (but with some commas like `"{, , ,}"` but none of these were satisfying.